windows/src/developer/kmcmpdll/Compiler.cpp

```diff
@@ -2494,6 +2496,13 @@ DWORD GetXString(PFILE_KEYBOARD fk, PWSTR str, PWSTR token, PWSTR output, int ma
         err = process_set_synonym(TSS_LAYER, fk, q, tstr, &mx);
         if (err != CERR_None) return err;
         continue;
+      case 19:  // #2241
+        if (*(p + 1) != '.') return CERR_InvalidToken;
+        if (sFlag) return CERR_InvalidInVirtualKeySection;
+        p += 2;
+        err = process_expansion(fk, p, tstr, &mx, max);
```

Comment:
I noticed the param `p` becomes `q` in l. 2675

Would this be valid?
```suggestion
        q = p+2;
        err = process_expansion(fk, q, tstr, &mx, max);
```

Reply:
No, that wouldn't work. `p` is our working variable there. We could rename the parameter in `process_expansion` though.